### PR TITLE
CORE-8020 Renaming HTTPRpc[verb] annotations, to Rest[verb] + other refactorings

### PR DESCRIPTION
**Pull request by pmck91**
This PR has mostly changed all of the various HTTP method annotations from RPC to HTTP and some other variable and class renaming. 

The final PR will be for the module name

### CODE REVIEW

File: libs/http-rpc/http-rpc-stubs-impl/src/main/kotlin/net/corda/httprpc/endpoints/impl/HelloRestResourceImpl.kt [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-import net.corda.httprpc.annotations.HttpRpcPOST
-import net.corda.httprpc.annotations.HttpRpcQueryParameter
+import net.corda.httprpc.annotations.HttpPOST
+import net.corda.httprpc.annotations.RestQueryParameter
 import net.corda.httprpc.security.CURRENT_RPC_CONTEXT
 import net.corda.v5.base.util.contextLogger
 import org.osgi.service.component.annotations.Component
 
-@HttpRpcResource(name = "Hello RPC API", description = "The Hello RPC API is used to test interactions via the " +
+@HttpRestResource(name = "Hello RPC API", description = "The Hello RPC API is used to test interactions via the " +
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc-stubs-impl/src/main/kotlin/net/corda/httprpc/endpoints/impl/HelloRestResourceImpl.kt:12]: RPC is mentioned here. I'll be publishing a new PR shortly, I'll tackle it here.

File: libs/http-rpc/http-rpc-test-common/src/main/kotlin/net/corda/httprpc/test/CalendarRestResource.kt [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 import net.corda.httprpc.RestResource
-import net.corda.httprpc.annotations.HttpRpcPOST
-import net.corda.httprpc.annotations.HttpRpcResource
+import net.corda.httprpc.annotations.HttpPOST
+import net.corda.httprpc.annotations.HttpRestResource
 import net.corda.httprpc.durablestream.api.FiniteDurableCursorBuilder
 import java.time.DayOfWeek
 
-@HttpRpcResource(name = "CalendarRestResource", description = "Calendar RPC Ops", path = "calendar")
+@HttpRestResource(name = "CalendarRestResource", description = "Calendar RPC Ops", path = "calendar")
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc-test-common/src/main/kotlin/net/corda/httprpc/test/CalendarRestResource.kt:9]: RPC mentioned here.

File: libs/http-rpc/http-rpc/README.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
-# HTTP RPC API
+# Corda REST API
 
 Contains mainly annotations to be used on interface methods which need to be exposed for
-HTTP RPC remote calls. More information can be found
+Rest remote calls. More information can be found
 [here](https://engineering.r3.com/product-areas/corda-platform/rpc-and-node-interaction/docs/http-rpc/developer/exposing-rpcops-functionality/annotations/).
```
Comment (ben-millar) [inline libs/http-rpc/http-rpc/README.md:5]: I can't reach this URL. Agreed offline that @pmck91 will notify docs team on this.

**Review by ben-millar — APPROVED** [review]
LGTM provided build passes; noted some issues, @ben-millar will make comment changes, @pmck91 will inform docs team of outdated links and follow up.

File: libs/http-rpc/http-rpc-stubs-impl/src/main/kotlin/net/corda/httprpc/endpoints/impl/HelloRestResourceImpl.kt (hunk 1 above)
Comment (ben-millar) [reply libs/http-rpc/http-rpc-stubs-impl/src/main/kotlin/net/corda/httprpc/endpoints/impl/HelloRestResourceImpl.kt:12]: NVM, this is contained in `http-rpc-stubs-impl` which will be handled in [CORE-8027](https://r3-cev.atlassian.net/browse/CORE-8027).

File: libs/http-rpc/http-rpc/README.md (hunk 3 above)
Comment (pmck91) [reply libs/http-rpc/http-rpc/README.md:5]: I have left a comment on the docs ticket associated with this ticket

File: libs/http-rpc/http-rpc-test-common/src/main/kotlin/net/corda/httprpc/test/CalendarRestResource.kt (hunk 2 above)
Comment (ben-millar) [reply libs/http-rpc/http-rpc-test-common/src/main/kotlin/net/corda/httprpc/test/CalendarRestResource.kt:9]: This is contained in `http-rpc-test-common` and will be handled in [CORE-8028](https://r3-cev.atlassian.net/browse/CORE-8028)! Tutto bene

**Pull request merged**